lua/ui/game/announcement/AbstractAnnouncement.lua

```diff
@@ -0,0 +1,421 @@
+            end
+        )
+
+        background.Right:Set(
+            function() 
+                local percentage = progress()
+                return percentage * content.Right() + (1 - percentage) * control.Right()
+            end
+        )
+
+        -- define width and height by top/bottom and left/right values
+        Layouter(background)
+            :ResetWidth()
+            :ResetHeight()
+            :End()
+
+        -- animate it
+        local startTime = GetSystemTimeSeconds()
+        local endTime = startTime + duration
+        while not IsDestroyed(self) do
+            local currentTime = GetSystemTimeSeconds()
+            if currentTime > endTime then
+                break
+            end
+
+            -- compute where we are in the animation
+            local percentage = math.clamp((currentTime - startTime) / (endTime - startTime), 0, 1)
+            if state == 'Contract' then
+                percentage = 1 - percentage
+            end
+
+            -- triggers the translation and scaling changes
+            progress:Set(percentage)
+
+            -- update alpha separately
+            background:SetAlpha(percentage, true)
+
+            WaitFrames(1)
+        end
+
+        -- make sure animation completes properly even when there are frame drops
+        if state == 'Expand' then
+            progress:Set(1)
+        else
+            progress:Set(0)
+        end
+    end,
+
+    --#endregion
+
+    ---------------------------------------------------------------------------
+    --#region Interface to abort 
+
+    --- Aborts the announcement. All animations are stopped. The announcement is turned transparent and then destroyed.
+    ---@param self UIAbstractAnnouncement
+    AbortAnnouncement = function(self)
+        -- make the function idempotent
+        if self.AbortAnimationThreadInstance then
+            return
+        end
+
+        self:CancelAnimation()
+        self:CancelContentAnimation()
+
+        self.AbortAnimationThreadInstance = self.Trash:Add(ForkThread(self.AbortAnimationThread, self))
+    end,
+
+    --- Aborts the announcement by turning it completely transparent. The announcement is destroyed at the end of it.
+    ---@param self UIAbstractAnnouncement
+    AbortAnimationThread = function(self)
+        local alphaContent = self.ContentArea:GetAlpha()
+        local alphaBackground = self.Background:GetAlpha()
+
+        while not IsDestroyed(self) do
+            alphaContent = math.clamp(alphaContent - 0.05, 0, 1)
+            alphaBackground = math.clamp(alphaBackground - 0.05, 0, 1)
+
+            self.ContentArea:SetAlpha(alphaContent, true)
+            self.Background:SetAlpha(alphaBackground, true)
+
+            if alphaContent == 0 and alphaBackground == 0 then
+                break
+            end
+
+            WaitFrames(1)
+        end
+
+        self:Destroy()
+    end,
+
+    --- Cancels the animation thread.
```

Comment:
I would call it main because it executes both the background and content animations.
```suggestion
    --- Cancels the main animation thread.
```